packages/core/admin/_internal/node/rspack/config.ts

```diff
@@ -0,0 +1,231 @@
+import { HotModuleReplacementPlugin, RspackOptions, RuleSetRule } from '@rspack/core';
+// @ts-expect-error – no types
+import ReactRefreshPlugin from '@rspack/plugin-react-refresh';
+import path from 'node:path';
+
+import { loadStrapiMonorepo } from '../core/monorepo';
+import type { BuildContext } from '../createBuildContext';
+import { getUserConfig } from '../core/config';
+import { getMonorepoAliases } from '../core/aliases';
+import ForkTsCheckerWebpackPlugin from 'fork-ts-checker-webpack-plugin';
+import findRoot from 'find-root';
+
+const resolveBaseConfig = async (ctx: BuildContext) => {
+  const monorepo = await loadStrapiMonorepo(ctx.cwd);
+
+  return {
+    entry: {
+      main: [`./${ctx.entry}`],
+    },
+    resolve: {
+      alias: {
+        ...getMonorepoAliases({ monorepo }),
+        'react/jsx-runtime.js': path.join(findRoot(require.resolve('react')), 'jsx-runtime'),
+      },
+    },
+    module: {
+      rules: [
+        {
+          test: /\.(svg|eot|otf|ttf|woff|woff2)$/,
+          type: 'asset/resource',
+        },
+        {
+          test: [/\.bmp$/, /\.gif$/, /\.jpe?g$/, /\.png$/, /\.ico$/],
+          type: 'asset/resource',
+          parser: {
+            dataUrlCondition: {
+              maxSize: 1000,
+            },
+          },
+        },
+        {
+          test: /\.(mp4|webm)$/,
+          type: 'asset/resource',
+          parser: {
+            dataUrlCondition: {
+              maxSize: 10000,
+            },
+          },
+        },
+      ],
+    },
+    target: ['web'],
+    builtins: {
+      define: {
```

Comment:
notice: Rspack already supports [definePlugin](https://www.rspack.dev/config/plugins.html#defineplugin), so definePlugin is preferred